sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.aest.controller.associateLegacyStudent.e2e-spec.ts

```diff
@@ -0,0 +1,232 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+  AESTGroups,
+  getAESTUser,
+} from "../../../../testHelpers";
+import {
+  saveFakeStudent,
+  saveFakeSFASIndividual,
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeUser,
+  createFakeSFASRestriction,
+} from "@sims/test-utils";
+import * as faker from "faker";
+import { getISODateOnlyString } from "@sims/utilities";
+import { NoteType } from "@sims/sims-db";
+
+/**
+ * Creates a random SIN with the intention to be unique and not interfere in other tests
+ */
+const UNIQUE_SIN = "#1234567#";
+/**
+ * Creates a random SFAS individual ID with the intention to be unique
+ * and potentially not interfere in other tests.
+ * SFAS IDs are not generated automatically.
+ */
+const UNIQUE_SFAS_RESTRICTION_ID = 999955551;
+
+describe("StudentAESTController(e2e)-associateLegacyStudent", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  beforeEach(async () => {
+    await Promise.all([
+      db.sfasIndividual.delete({ sin: UNIQUE_SIN }),
+      db.sfasRestriction.delete({ id: UNIQUE_SFAS_RESTRICTION_ID }),
+    ]);
+  });
+
+  it(
+    "Should associate a student to a legacy profile, create a note, and set restriction as processed false" +
+      " when the student is not associated yet to a legacy profile and a legacy profile is a potential match.",
+    async () => {
+      // Arrange.
+      const user = createFakeUser();
+      const userLastName = faker.datatype.uuid();
+      user.lastName = userLastName;
+      const birthDate = getISODateOnlyString(new Date());
+      const student = await saveFakeStudent(
+        db.dataSource,
+        { user },
+        { initialValue: { birthDate } },
+      );
+      const legacyProfileMatch = await saveFakeSFASIndividual(db.dataSource, {
+        initialValues: { lastName: userLastName, birthDate },
+      });
+      const processedSFASRestriction = createFakeSFASRestriction({
+        initialValues: {
+          id: UNIQUE_SFAS_RESTRICTION_ID,
+          individualId: legacyProfileMatch.id,
+          processed: true,
+        },
+      });
+      await db.sfasRestriction.save(processedSFASRestriction);
+
+      const aestUserToken = await getAESTToken(
+        AESTGroups.BusinessAdministrators,
+      );
+      const ministryUser = await getAESTUser(
+        db.dataSource,
+        AESTGroups.BusinessAdministrators,
+      );
+      const payload = {
+        individualId: legacyProfileMatch.id,
+        noteDescription: faker.datatype.uuid(),
```

Comment:
Minor and not a blocker. Note desc can be a `faker.random.alpha`

Reply:
The reason for using it was to ensure uniqueness only.